### PR TITLE
Fix/override kwargs text parser block

### PR DESCRIPTION
**Pull request by mihirathale98**
This PR enables to functionality to override kwargs for TextParserBlock type during runtime

#### Features 
1. Pass runtime args for TextParserBlock type as well in `flow.generate()` method
2. Enables custom parsing logic for each block at runtime, ease of use for the users
3. Prevents the hassle of changing the config manually in the `flow.yaml` file

#### Usage 

```python

flow_path = FlowRegistry.get_flow_path('small-rock-799')
flow = Flow.from_yaml(flow_path)

dataset = your_dataset

runtime_params = {
    "parse_atomic_facts" : {
        "save_reasoning_content": True
        }  
}

result = flow.generate(dataset, runtime_params=runtime_params)

```

#### Files Changed

1. `src/sdg_hub/core/blocks/llm/text_parser_block.py` - Added logic to override args 
2. `tests/blocks/llm/test_textparserblock.py` - tests for the override functionality 

## Summary by CodeRabbit

* **New Features**
  * Per-call runtime overrides for parsing (pattern, start/end tags), list expansion, and reasoning-content capture with optional custom field name; unknown override keys are warned and ignored and overrides apply only to the invocation.

* **Refactor**
  * Runtime overrides are validated/normalized (including tag/count and reasoning-field checks), logged at runtime, and original settings are reliably restored after execution.

* **Tests**
  * Extensive tests for override validation, application/restoration, reasoning-content behavior, single/list inputs, edge cases, and logging.

### CODE REVIEW

File: src/sdg_hub/core/blocks/llm/text_parser_block.py [hunk 1]
```diff
@@ -407,13 +406,26 @@ def _generate(self, sample: dict) -> list[dict]:
...
-            new_data.extend(self._generate(sample))
-        return Dataset.from_list(new_data)
+    def generate(self, samples: Dataset, **override_kwargs: Any) -> Dataset:
+        # Apply runtime parameter overrides
+        original_values: dict[str, Any] = {}
+        for param_name, param_value in override_kwargs.items():
+            if hasattr(self, param_name):
+                original_values[param_name] = getattr(self, param_name)
+                setattr(self, param_name, param_value)
+                logger.debug(f"Runtime override: {param_name} = {param_value}")
```
[the remark this answers is not in the document]
Comment (mihirathale98) [reply src/sdg_hub/core/blocks/llm/text_parser_block.py]: Added validation

File: src/sdg_hub/core/blocks/llm/text_parser_block.py [hunk 2]
```diff
@@ -407,13 +433,131 @@ def _generate(self, sample: dict) -> list[dict]:
             )
             return []
 
-    def generate(self, samples: Dataset, **kwargs: Any) -> Dataset:
-        logger.debug(f"Parsing outputs for {len(samples)} samples")
-        if len(samples) == 0:
-            logger.warning("No samples to parse, returning empty dataset")
-            return Dataset.from_list([])
+    def _validate_override_kwargs(
```
Comment (eshwarprasadS) [inline src/sdg_hub/core/blocks/llm/text_parser_block.py:436]: Do we need a new validate method here? could we explore the path of calling `_validate` or `_validate_custom` after applying overrides? duplicated validation logic will be hard to maintain.
